context/AuthContextProvider.tsx

```diff
@@ -68,14 +67,11 @@ export const AuthContextProvider = ({
 
   /**
    * Log out and clear session state
-   * @returns {Promise<void>}
+   * @returns {Promise<void | Error>}
    */
-  const logoutAccount = async (): Promise<void> => {
+  const logoutAccount = async (): Promise<void | Error> => {
     try {
-      await account.deleteSession('current');
-      setIsSignedIn(false);
-      resetUser(); // Reset user data in the store
-      router.push('/login');
+      await logoutFunction({ setIsSignedIn, router, resetUser });
     } catch (error) {
       console.error('Logout error:', error);
```

Comment:
This should fire if an error occurred from logoutFunction.

Reply:
It's ok to have this. What I'm asking is how is this catching the error if the logoutFunction() is not throwing an error? 
https://github.com/LetsGetTechnical/gridiron-survivor/pull/385/files#diff-4dbc1b91d9d9b4016139ec38778a854b975c876526505f87077bce86729e2480R78

Reply:
it's not throwing an error. fixed by deleting try catch